spec/solver_spec.rb

```diff
@@ -0,0 +1,46 @@
+require_relative '../solver'
+
+describe Solver do
+  context 'Factorial' do
+    it 'takes one argument, an integer N, and returns the factorial for that number' do
```

Comment:
- [ ] You should also test for the Special case, when input is 0, the result should equal 1, kindly consider implementing this